csvcubed/csvcubed/readers/v1_0/columnschema.py

```diff
@@ -0,0 +1,459 @@
+"""
+import uritemplate
+
+from dataclasses import dataclass
+from typing import List, Union, Optional, TypeVar
+from pathlib import Path
+
+from csvcubed.models.cube import (
+    CatalogMetadata,
+    CompositeQbCodeList,
+    DuplicatedQbConcept,
+    ExistingQbAttributeLiteral,
+    ExistingQbCodeList,
+    NewQbAttributeLiteral,
+    NewQbCodeList,
+    NewQbCodeListInCsvW, QbCodeList,
+)
+
+from csvcubedmodels.dataclassbase import DataClassBase
+
+from csvcubed.inputs import pandas_input_to_columnar_optional_str
+from csvcubed.models.cube.qb.components import (
+    NewQbDimension,
+    ExistingQbDimension,
+    NewQbAttribute,
+    NewQbAttributeValue,
+    ExistingQbAttribute,
+    NewQbUnit,
+    ExistingQbUnit,
+    QbMultiUnits,
+    QbMultiMeasureDimension,
+    QbMultiMeasureObservationValue,
+    QbSingleMeasureObservationValue,
+    ExistingQbMeasure,
+    NewQbMeasure,
+    QbObservationValue,
+)
+from csvcubed.inputs import PandasDataTypes
+from csvcubed.utils.uri import (
+    csvw_column_name_safe,
+    looks_like_uri,
+)
+
+T = TypeVar("T", bound=object)
+
+
+@dataclass
+class SchemaBaseClass(DataClassBase):
+    ...
+
+
+@dataclass
+class NewDimension(SchemaBaseClass):
+    # Schema property - but removed for json to dataclass mapping
+    # type: # str = "dimension"
+
+    # Properties only available for New Dimension
+    label: str = ""
+    description: Optional[str] = None
+    definition_uri: Optional[str] = None
+
+    # Properties common to both New and Existing Dimension
+    from_existing: Optional[str] = None
+    cell_uri_template: Optional[str] = None
+    code_list: Optional[Union[str, bool]] = True
+
+    def map_to_new_qb_dimension(
+        self, label: str, data: PandasDataTypes, json_parent_dir: Path
+    ) -> NewQbDimension:
+
+        new_dimension = NewQbDimension.from_data(
+            label=self.label or label,
+            data=data,
+            description=self.description,
+            parent_dimension_uri=self.from_existing,
+            source_uri=self.definition_uri,
+        )
+        new_dimension.code_list = self._get_code_list(new_dimension, json_parent_dir)
+        return new_dimension
+
+    def _get_code_list(
+        self,
+        new_dimension: NewQbDimension,
+        json_parent_dir: Path,
+    ) -> Optional[QbCodeList]:
+
+        code_list_obj = None
+
+        if isinstance(self.code_list, str):
+            if looks_like_uri(self.code_list):
+                code_list_obj = ExistingQbCodeList(self.code_list)
+
+            else:
+                code_list_path = Path(self.code_list)
+                if code_list_path.is_absolute():
+                    code_list_obj = NewQbCodeListInCsvW(code_list_path)
+                else:
+                    code_list_obj = NewQbCodeListInCsvW(
+                        json_parent_dir / self.code_list
+                    )
+        elif isinstance(self.code_list, bool):
+            if self.code_list is False:
```

Comment:
This is (I beleive) in line with info json, I've added a comment to help.
# The NewQbCodeList and Concepts are populated in the NewQbDimension.from_data() call
# the _get_code_list method overrides the code_list if required.

Perhaps we want to change the logic so that the NewQbCodeList is created within the _get_code_list method. - Please confirm.

Reply:
https://github.com/GSS-Cogs/gss-utils/blob/master/gssutils/csvcubedintegration/configloaders/infojson1point1/columnschema.py#L70

So it looks like we are missing some logic (see above link) which supports the `"code_list": true` behaviour.

Reply:
I've added this in.